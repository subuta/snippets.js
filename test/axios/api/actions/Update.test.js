/* global expect, describe, it */

import { build, format, snippets as s } from 'bld.js'

import UpdateAction from 'lib/axios/api/actions/Update'

import {
  Routes as RoutesConfig,
  Models as ModelsConfig
} from 'test/fixtures/config'

describe('axios/api/actions/Update', () => {
  it('should Update axios Api Update action with channel', () => {
    const channelRoute = RoutesConfig['channel']
    const channelModel = ModelsConfig['channel']

    expect(format(UpdateAction(channelRoute, channelModel))).toMatchSnapshot()
  })

  it('should Update axios Api Update action with comment', () => {
    const channelRoute = RoutesConfig['comment']
    const channelModel = ModelsConfig['comment']

    expect(format(UpdateAction(channelRoute, channelModel))).toMatchSnapshot()
  })
})
