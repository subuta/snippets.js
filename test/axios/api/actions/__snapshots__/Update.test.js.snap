// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`axios/api/actions/Update should Update axios Api Update action with channel 1`] = `
"export const update = (id, params) => {
  return request.put(\`/channels/\${id}\`, {
    channel: params
  })
}
"
`;

exports[`axios/api/actions/Update should Update axios Api Update action with comment 1`] = `
"export const update = (id, params) => {
  return request.put(\`/channels/\${params.channelId}/comments/\${id}\`, {
    comment: params
  })
}
"
`;
